.circleci/integration-tests/master_dag.py

```diff
@@ -242,6 +242,18 @@ def prepare_dag_dependency(task_info, execution_time):
         provide_context=True,
     )
 
+    end_test = DummyOperator(
+        task_id="end_test",
+        trigger_rule="all_success",
```

Comment:
The nuke DAG may not run as it is a dependency of this task and this task will only be triggered when all previous DAGs are successful

Reply:
we have added a trigger rule in`TriggerDagRunOperator` isn't that sufficient?

Reply:
No, it will not be sufficient. end_test will not be triggered if any of the DAGs fail. And in that case, since end_test will not be triggered, it will never be done and hence nuke DAG won't be triggered.

Reply:
if the previous DAG will fail the status of `end_test` will be upstream fail, isn't that the case?
and if a task has trigger_rule "all_done" it always runs (which is the case of the nuke task)? or I'm missing something

Reply:
can you please paste a screenshot of the Graph when one of the providers DAGs fail? If `upstream_failed` tasks do not hinder `all_done` then it I guess you're right and we're safe. But will the final `end` task report an `upstream_failed` when one of the providers' DAGs fails? Or will it be marked as a success if only the nuke DAG runs successfully? My questions is because we should mark the master DAG status as failed if any of the DAGs fail.

Reply:
ok, I'll put a screenshot here but it should be `[provider-DAG]->report->end_test->nuke->end`

Reply:
yes and try to fail the provider DAG on purpose to observe the behaviour

Reply:
<img width="675" alt="Screenshot 2023-05-02 at 4 26 17 PM" src="https://user-images.githubusercontent.com/98807258/235648341-0ae49fd7-86bb-4ef1-bb55-b9e0ac144799.png">

Reply:
Why did the aws_nuke_dag fail?

what happens if provider DAGs fail but `aws_nuke_dag` succeeds? Does the `end` task report upstream_failed? Because if `end` is marked as success in that case then the `master_dag` will also be marked as successful, no?

Reply:
if we think that way we have more problems. what happens if all tasks succeed and bigquery fail? the task will be marked a success. the trigger rule applies to just the previous task and DAG status is the status of the last task.  but I'm ok to remove the end task but that will not solve the problem you are pointing out. One thing  we could do is make all task previous to the end task but that looks out of the scope of this PR

Reply:
The problem I see is the change in this PR might alter the behaviour we have currently in the `main` branch.

Currently, if any provider DAGs fail then the master DAG is also marked as `failed`. But with this change, I think if only the nuke dag runs fine which could & should happen even if any provider DAG fails, then the `end` will be marked `success` and also the `master_dag` will be marked `success` **even though provider DAGs might have failed**.

Attaching screenshots of the current DAG graph and how we mark the `master_dag` as `failed` even if one provider DAG fails.
<img width="1232" alt="Screenshot 2023-05-02 at 4 52 18 PM" src="https://user-images.githubusercontent.com/10206082/235653339-b422a9e1-fbe0-4ca9-8e7e-762501e16b58.png">

<img width="750" alt="Screenshot 2023-05-02 at 4 52 50 PM" src="https://user-images.githubusercontent.com/10206082/235653357-e2246aaf-b91f-4311-bc37-d41e29763b5c.png">

Reply:
I see the end task is success in the graph view but upstream fail in the grid view, is that expected?

Reply:
yes, that is another issue but that fix can come in a separate PR in my opinion. But, at least we should try to keep the existing behaviour as it is in this PR. WDYT?

Reply:
I have updated the DAG a little bit but still, it would not solve the problem that you are pointing out for all cases.

<img width="563" alt="Screenshot 2023-05-02 at 5 38 27 PM" src="https://user-images.githubusercontent.com/98807258/235662114-bad2d9e2-27c8-44c2-b523-ad2758b28813.png">

Reply:
this looks like a good idea to me for now. The other issue is an existing issue and we can address that in another PR later